File: typescript-client-sdk/src/sdk/models/shared/codatdataintegritycontractsmetadatamatchdateinfo.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform } from "class-transformer";


export class CodatDataIntegrityContractsMetadataMatchDateInfo extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "maxDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  maxDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "maxOverlappingDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  maxOverlappingDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "minDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  minDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "minOverlappingDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  minOverlappingDate?: Date;
}

File: go-client-sdk/pkg/models/shared/codatdataintegritycontractsmetadatamatchdateinfo.go
package shared

import (
	"time"
)

type CodatDataIntegrityContractsMetadataMatchDateInfo struct {
	MaxDate            *time.Time `json:"maxDate,omitempty"`
	MaxOverlappingDate *time.Time `json:"maxOverlappingDate,omitempty"`
	MinDate            *time.Time `json:"minDate,omitempty"`
	MinOverlappingDate *time.Time `json:"minOverlappingDate,omitempty"`
}
